core/src/avm1/globals/movie_clip.rs

```diff
@@ -121,6 +122,62 @@ pub fn constructor<'gc>(
     Ok(this.into())
 }
 
+fn new_rectangle<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    rectangle: Rectangle,
+) -> Result<Value<'gc>, Error<'gc>> {
+    let x = rectangle.x_min.to_pixels();
+    let y = rectangle.y_min.to_pixels();
+    let width = (rectangle.x_max - rectangle.x_min).to_pixels();
+    let height = (rectangle.y_max - rectangle.y_min).to_pixels();
+    let args = &[x.into(), y.into(), width.into(), height.into()];
+    let proto = activation.context.avm1.prototypes.rectangle;
+    proto.construct(activation, args)
+}
+
+fn object_to_rectangle<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    object: Object<'gc>,
+) -> Result<Rectangle, Error<'gc>> {
+    let x = object.get("x", activation)?.coerce_to_i32(activation)?;
+    let y = object.get("y", activation)?.coerce_to_i32(activation)?;
+    let width = object.get("width", activation)?.coerce_to_i32(activation)?;
```

Comment:
Looks like these should be `get_local_stored`, and if any of the fields return `None`, the previous scroll rect is unchanged, or a zeroed rectangle is set if there was no previous scroll rect.

```
import flash.geom.Rectangle;


test.scrollRect = {x: 5};
trace(test.scrollRect); // (x=0, y=0, w=0, h=0)

var o = {
	x: {valueOf:function() { trace("VO1"); return 1; }},
	y: {valueOf:function() { trace("VO2"); return 2; }},
	width: {valueOf:function() { trace("VO3"); return 3; }},
	height: {valueOf:function() { trace("VO4"); return 4; }}
};
test.scrollRect = o;
trace(test.scrollRect); // (x=1, y=2, w=3, h=4)

o = {
	x: {valueOf:function() { trace("VO1"); return 1; }},
	// y: {valueOf:function() { trace("VO2"); return 2; }},
	width: {valueOf:function() { trace("VO3"); return 3; }},
	height: {valueOf:function() { trace("VO4"); return 4; }}
};
o.__proto__.y = 99;
trace(test.scrollRect); // (x=1, y=2, w=3, h=4), unchanged
```

Reply:
Good catch! In fact, this gets much weirder when you unset `scrollRect` then "set" it back:
```as
test.scrollRect = {x: 1, y: 2, width: 3, height: 4};
trace(test.scrollRect); // (x=1, y=2, w=3, h=4)

test.scrollRect = undefined;
trace(test.scrollRect); // undefined

test.scrollRect = {x: 1};
trace(test.scrollRect); // (x=1, y=2, w=3, h=4)
```
The first `scrollRect` is remembered as if Flash stores a boolean of "whether a scroll rect is set" separately from the scroll rect itself. I guess we should do the same instead of `Option<Rectangle>`.